### PR TITLE
Add actions to CavcRemandProcessedLetterResponseWindowTask

### PR DESCRIPTION
**Pull request by hschallhorn**
Bumps [caseflow-112](https://vajira.max.gov/browse/CASEFLOW-112)

Part 2 of stack
Part 1: #15697

### Description
Adds admin actions to `CavcRemandProcessedLetterResponseWindowTask`. Allows users and admins to assign/reassign the task. Allows users to make the task complete and make the case available for distribution.

### Acceptance Criteria
- [ ] The following actions are added to `CavcRemandProcessedLetterResponseWindowTask` and can be completed by an admin or the assignee of the task
  - [ ] Mark task complete
    - [ ] Closes self and parent to make appeal ready for distribution
  - [ ] FOIA - sent to privacy team
    - [ ] Creates a privacy act task assigned to the privacy team as a child of the distribution task
  - [ ] Translation - sent translation team
    - [ ] Creates a translation task assigned to the translation team as a child of the distribution task
  - [ ] Missing Hearing Transcripts - sent to hearing branch
    - [ ] Creates a transcription task assigned to the transcription team as a child of the distribution task
  - [ ] Send to VSO - Sent to colocated team
    - [ ] Creates an IHP task assigned to the colocated team as a child of the distribution task
  - [ ] Send to Hearings team
    - [ ] Creates a schedule hearing task assigned to Bva as a child of a hearing task (that is the child of the distribution task)
- [ ] Admins can assign the task to a user on their team
- [ ] The assignee of the task can reassign it to a user on their team

### Testing Plan
1. Log in as CAVC_LIT_SUPPORT_ADMIN and go to http://localhost:3000/organizations/cavc-lit-support
1. Open a few Send CAVC Letter tasks in new tabs, assign them to yourself, complete them, and end the hold on the new CavcRemandProcessedLetterResponseWindowTask

#### Ensure the admin can assign a task to a user on the team
1. Select "assign to user"
1. Select yourself
1. Add instructions
1. Press submit
1. Confirm success banner, task now shows in your queue, case details shows instructions
```ruby
                                                                ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
Appeal 920 (E 201213-919 Court Remand) ──────────────────────── │ ID   │ STATUS      │ ASGN_BY                │ ASGN_TO                │ UPDATED_AT              │
└── RootTask                                                    │ 2733 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
    └── DistributionTask                                        │ 2734 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        └── CavcTask                                            │ 2735 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
            ├── SendCavcRemandProcessedLetterTask               │ 2736 │ completed   │                        │ CavcLitigationSupport  │ 2020-12-14 20:26:56 UTC │
            │   └── SendCavcRemandProcessedLetterTask           │ 5766 │ completed   │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_ADMIN │ 2020-12-14 20:26:56 UTC │
            └── CavcRemandProcessedLetterResponseWindowTask     │ 5772 │ on_hold     │                        │ CavcLitigationSupport  │ 2020-12-14 20:50:25 UTC │
                ├── TimedHoldTask                               │ 5773 │ cancelled   │                        │ CavcLitigationSupport  │ 2020-12-14 20:27:42 UTC │
                └── CavcRemandProcessedLetterResponseWindowTask │ 5787 │ assigned    │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_ADMIN │ 2020-12-14 20:50:58 UTC │
                                                                └────────────────────────────────────────────────────────────────────────────────────────────────┘
```

#### Ensure assignee can reassign task to another user
1. Select "reassign to user"
1. Select another team memeber
1. Add instructions
1. Press submit
1. Confirm success banner, task no longer shows in your queue, case details shows instructions and new assignee
```ruby
                                                                ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
Appeal 920 (E 201213-919 Court Remand) ──────────────────────── │ ID   │ STATUS      │ ASGN_BY                │ ASGN_TO                │ UPDATED_AT              │
└── RootTask                                                    │ 2733 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
    └── DistributionTask                                        │ 2734 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        └── CavcTask                                            │ 2735 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
            ├── SendCavcRemandProcessedLetterTask               │ 2736 │ completed   │                        │ CavcLitigationSupport  │ 2020-12-14 20:26:56 UTC │
            │   └── SendCavcRemandProcessedLetterTask           │ 5766 │ completed   │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_ADMIN │ 2020-12-14 20:26:56 UTC │
            └── CavcRemandProcessedLetterResponseWindowTask     │ 5772 │ on_hold     │                        │ CavcLitigationSupport  │ 2020-12-14 20:50:25 UTC │
                ├── TimedHoldTask                               │ 5773 │ cancelled   │                        │ CavcLitigationSupport  │ 2020-12-14 20:27:42 UTC │
                ├── CavcRemandProcessedLetterResponseWindowTask │ 5787 │ cancelled   │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_ADMIN │ 2020-12-14 20:50:58 UTC │
                └── CavcRemandProcessedLetterResponseWindowTask │ 5788 │ in_progress │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_USER1 │ 2020-12-14 20:50:58 UTC │
                                                                └────────────────────────────────────────────────────────────────────────────────────────────────┘
```
#### Ensure assignee can create Translation admin actions
1. Find a new case
1. Click the task action dropdown and confirm new "send to translation team" action
1. Confirm you can add instructions to this action
1. Confirm clicking submit keeps you on the case details page and creates a translation task assigned to the translation team is created as a child of the distribution task
1. Confirm you still have actions available to reassign or complete
1. Confirm you cannot create another task assigned to the translation team
```ruby
                                                                ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
Appeal 920 (E 201213-919 Court Remand) ──────────────────────── │ ID   │ STATUS      │ ASGN_BY                │ ASGN_TO                │ UPDATED_AT              │
└── RootTask                                                    │ 2733 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
    └── DistributionTask                                        │ 2734 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        ├── CavcTask                                            │ 2735 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        │   ├── SendCavcRemandProcessedLetterTask               │ 2736 │ completed   │                        │ CavcLitigationSupport  │ 2020-12-14 20:26:56 UTC │
        │   │   └── SendCavcRemandProcessedLetterTask           │ 5766 │ completed   │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_ADMIN │ 2020-12-14 20:26:56 UTC │
        │   └── CavcRemandProcessedLetterResponseWindowTask     │ 5772 │ on_hold     │                        │ CavcLitigationSupport  │ 2020-12-14 20:50:25 UTC │
        │       ├── TimedHoldTask                               │ 5773 │ cancelled   │                        │ CavcLitigationSupport  │ 2020-12-14 20:27:42 UTC │
        │       └── CavcRemandProcessedLetterResponseWindowTask │ 5788 │ in_progress │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_USER1 │ 2020-12-14 20:50:58 UTC │
        └── TranslationTask                                     │ 5793 │ assigned    │ CAVC_LIT_SUPPORT_ADMIN │ Translation            │ 2020-12-15 14:13:34 UTC │
                                                                └────────────────────────────────────────────────────────────────────────────────────────────────┘
```

#### Ensure assignee can create Transcription admin actions
1. Click the task action dropdown and confirm new "send to transcription team" action
1. Confirm you can add instructions to this action
1. Confirm clicking submit keeps you on the case details page and creates a transcription task assigned to the trascription team is created as a child of the distribution task
1. Confirm you still have actions available to reassign or complete
1. Confirm you cannot create another task assigned to the trancription team
```ruby
                                                                ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
Appeal 920 (E 201213-919 Court Remand) ──────────────────────── │ ID   │ STATUS      │ ASGN_BY                │ ASGN_TO                │ UPDATED_AT              │
└── RootTask                                                    │ 2733 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
    └── DistributionTask                                        │ 2734 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        ├── CavcTask                                            │ 2735 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        │   ├── SendCavcRemandProcessedLetterTask               │ 2736 │ completed   │                        │ CavcLitigationSupport  │ 2020-12-14 20:26:56 UTC │
        │   │   └── SendCavcRemandProcessedLetterTask           │ 5766 │ completed   │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_ADMIN │ 2020-12-14 20:26:56 UTC │
        │   └── CavcRemandProcessedLetterResponseWindowTask     │ 5772 │ on_hold     │                        │ CavcLitigationSupport  │ 2020-12-14 20:50:25 UTC │
        │       ├── TimedHoldTask                               │ 5773 │ cancelled   │                        │ CavcLitigationSupport  │ 2020-12-14 20:27:42 UTC │
        │       └── CavcRemandProcessedLetterResponseWindowTask │ 5788 │ in_progress │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_USER1 │ 2020-12-14 20:50:58 UTC │
        └── TranscriptionTask                                   │ 5794 │ assigned    │ CAVC_LIT_SUPPORT_ADMIN │ TranscriptionTeam      │ 2020-12-15 14:15:11 UTC │
                                                                └────────────────────────────────────────────────────────────────────────────────────────────────┘
```

#### Ensure assignee can create  FOIA admin actions
1. Click the task action dropdown and confirm new "send to privacy team" action
1. Confirm you can add instructions to this action
1. Confirm clicking submit keeps you on the case details page and creates a privacy act task assigned to the privacy team is created as a child of the distribution task
1. Confirm you still have actions available to reassign or complete
1. Confirm you cannot create another task assigned to the privacy team
```ruby
                                                                ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
Appeal 920 (E 201213-919 Court Remand) ──────────────────────── │ ID   │ STATUS      │ ASGN_BY                │ ASGN_TO                │ UPDATED_AT              │
└── RootTask                                                    │ 2733 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
    └── DistributionTask                                        │ 2734 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        ├── CavcTask                                            │ 2735 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        │   ├── SendCavcRemandProcessedLetterTask               │ 2736 │ completed   │                        │ CavcLitigationSupport  │ 2020-12-14 20:26:56 UTC │
        │   │   └── SendCavcRemandProcessedLetterTask           │ 5766 │ completed   │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_ADMIN │ 2020-12-14 20:26:56 UTC │
        │   └── CavcRemandProcessedLetterResponseWindowTask     │ 5772 │ on_hold     │                        │ CavcLitigationSupport  │ 2020-12-14 20:50:25 UTC │
        │       ├── TimedHoldTask                               │ 5773 │ cancelled   │                        │ CavcLitigationSupport  │ 2020-12-14 20:27:42 UTC │
        │       └── CavcRemandProcessedLetterResponseWindowTask │ 5788 │ in_progress │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_USER1 │ 2020-12-14 20:50:58 UTC │
        └── PrivacyActTask                                      │ 5795 │ assigned    │ CAVC_LIT_SUPPORT_ADMIN │ PrivacyTeam            │ 2020-12-15 14:16:08 UTC │
                                                                └────────────────────────────────────────────────────────────────────────────────────────────────┘
```

#### Ensure assignee can create  IHP admin actions
1. Click the task action dropdown and confirm new "send ihp to colocated team" action
1. Confirm you can add instructions to this action
1. Confirm clicking submit keeps you on the case details page and creates an ihp act task assigned to the colocated team is created as a child of the distribution task. Colocated team auto assigns to its users so you should see a child task as a child of the org task
1. Confirm you still have actions available to reassign or complete
1. Confirm you cannot create another task assigned to the colocated team
```ruby
                                                                ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
Appeal 920 (E 201213-919 Court Remand) ──────────────────────── │ ID   │ STATUS      │ ASGN_BY                │ ASGN_TO                │ UPDATED_AT              │
└── RootTask                                                    │ 2733 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
    └── DistributionTask                                        │ 2734 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        ├── CavcTask                                            │ 2735 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        │   ├── SendCavcRemandProcessedLetterTask               │ 2736 │ completed   │                        │ CavcLitigationSupport  │ 2020-12-14 20:26:56 UTC │
        │   │   └── SendCavcRemandProcessedLetterTask           │ 5766 │ completed   │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_ADMIN │ 2020-12-14 20:26:56 UTC │
        │   └── CavcRemandProcessedLetterResponseWindowTask     │ 5772 │ on_hold     │                        │ CavcLitigationSupport  │ 2020-12-14 20:50:25 UTC │
        │       ├── TimedHoldTask                               │ 5773 │ cancelled   │                        │ CavcLitigationSupport  │ 2020-12-14 20:27:42 UTC │
        │       └── CavcRemandProcessedLetterResponseWindowTask │ 5788 │ in_progress │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_USER1 │ 2020-12-14 20:50:58 UTC │
        └── IhpColocatedTask                                    │ 5796 │ on_hold     │ CAVC_LIT_SUPPORT_ADMIN │ Colocated              │ 2020-12-15 14:16:56 UTC │
            └── IhpColocatedTask                                │ 5797 │ assigned    │ CAVC_LIT_SUPPORT_ADMIN │ BVALSPORER             │ 2020-12-15 14:16:56 UTC │
                                                                └────────────────────────────────────────────────────────────────────────────────────────────────┘
```

#### Ensure assignee can create Hearings admin actions
1. Click the task action dropdown and confirm new "send schedule hearing to hearings team team" action
1. Confirm you can add instructions to this action
1. Confirm clicking submit keeps you on the case details page and creates a schedule hearing task assigned to the BVA team is created as a child of a hearing task, which is a child of the the distribution task.
1. Confirm you still have actions available to reassign or complete
1. Confirm you cannot create another task assigned to the colocated team
```ruby
                                                                ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
Appeal 920 (E 201213-919 Court Remand) ──────────────────────── │ ID   │ STATUS      │ ASGN_BY                │ ASGN_TO                │ UPDATED_AT              │
└── RootTask                                                    │ 2733 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
    └── DistributionTask                                        │ 2734 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        ├── CavcTask                                            │ 2735 │ on_hold     │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        │   ├── SendCavcRemandProcessedLetterTask               │ 2736 │ completed   │                        │ CavcLitigationSupport  │ 2020-12-14 20:26:56 UTC │
        │   │   └── SendCavcRemandProcessedLetterTask           │ 5766 │ completed   │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_ADMIN │ 2020-12-14 20:26:56 UTC │
        │   └── CavcRemandProcessedLetterResponseWindowTask     │ 5772 │ on_hold     │                        │ CavcLitigationSupport  │ 2020-12-14 20:50:25 UTC │
        │       ├── TimedHoldTask                               │ 5773 │ cancelled   │                        │ CavcLitigationSupport  │ 2020-12-14 20:27:42 UTC │
        │       └── CavcRemandProcessedLetterResponseWindowTask │ 5788 │ in_progress │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_USER1 │ 2020-12-14 20:50:58 UTC │
        └── HearingTask                                         │ 5798 │ on_hold     │                        │ Bva                    │ 2020-12-15 14:18:09 UTC │
            └── ScheduleHearingTask                             │ 5799 │ assigned    │ CAVC_LIT_SUPPORT_ADMIN │ Bva                    │ 2020-12-15 14:18:09 UTC │
                                                                └────────────────────────────────────────────────────────────────────────────────────────────────┘
```

#### Ensure the user can complete their task
1. On the appeal with all the admin actions, select "complete task"
1. Ensure this completes the response window task and the parent cavc task
1. Ensure the distribution task is still on hold as it still has open children
1. Ensure the user no longer has actions on the appeal
```ruby
                                                                ┌──────────────────────────────────────────────────────────────────────────────────────────────┐
Appeal 920 (E 201213-919 Court Remand) ──────────────────────── │ ID   │ STATUS    │ ASGN_BY                │ ASGN_TO                │ UPDATED_AT              │
└── RootTask                                                    │ 2733 │ on_hold   │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
    └── DistributionTask                                        │ 2734 │ on_hold   │                        │ Bva                    │ 2020-12-14 17:47:23 UTC │
        ├── CavcTask                                            │ 2735 │ completed │                        │ Bva                    │ 2020-12-15 14:22:11 UTC │
        │   ├── SendCavcRemandProcessedLetterTask               │ 2736 │ completed │                        │ CavcLitigationSupport  │ 2020-12-14 20:26:56 UTC │
        │   │   └── SendCavcRemandProcessedLetterTask           │ 5766 │ completed │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_ADMIN │ 2020-12-14 20:26:56 UTC │
        │   └── CavcRemandProcessedLetterResponseWindowTask     │ 5772 │ completed │                        │ CavcLitigationSupport  │ 2020-12-15 14:22:11 UTC │
        │       ├── TimedHoldTask                               │ 5773 │ cancelled │                        │ CavcLitigationSupport  │ 2020-12-14 20:27:42 UTC │
        │       └── CavcRemandProcessedLetterResponseWindowTask │ 5788 │ completed │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_USER1 │ 2020-12-15 14:22:11 UTC │
        └── HearingTask                                         │ 5798 │ on_hold   │                        │ Bva                    │ 2020-12-15 14:18:09 UTC │
            └── ScheduleHearingTask                             │ 5799 │ assigned  │ CAVC_LIT_SUPPORT_ADMIN │ Bva                    │ 2020-12-15 14:18:09 UTC │
                                                                └──────────────────────────────────────────────────────────────────────────────────────────────┘
```
#### Ensure completing a task when the distribution task has no other children makes the case ready to distribute
1. Find another case
1. Complete the task
1. Ensure the distribution task is now "assigned" (the appeal is ready to be distributed)
```ruby
                                                            ┌──────────────────────────────────────────────────────────────────────────────────────────────┐
Appeal 924 (E 201213-923 Court Remand) ──────────────────── │ ID   │ STATUS    │ ASGN_BY                │ ASGN_TO                │ UPDATED_AT              │
└── RootTask                                                │ 2757 │ on_hold   │                        │ Bva                    │ 2020-12-14 17:47:25 UTC │
    └── DistributionTask                                    │ 2758 │ assigned  │                        │ Bva                    │ 2020-12-15 14:24:28 UTC │
        └── CavcTask                                        │ 2759 │ completed │                        │ Bva                    │ 2020-12-15 14:24:28 UTC │
            ├── SendCavcRemandProcessedLetterTask           │ 2760 │ completed │                        │ CavcLitigationSupport  │ 2020-12-14 20:27:04 UTC │
            │   └── SendCavcRemandProcessedLetterTask       │ 5768 │ completed │ CAVC_LIT_SUPPORT_ADMIN │ CAVC_LIT_SUPPORT_ADMIN │ 2020-12-14 20:27:04 UTC │
            └── CavcRemandProcessedLetterResponseWindowTask │ 5776 │ completed │                        │ CavcLitigationSupport  │ 2020-12-15 14:24:28 UTC │
                └── TimedHoldTask                           │ 5777 │ cancelled │                        │ CavcLitigationSupport  │ 2020-12-14 20:27:57 UTC │
                                                            └──────────────────────────────────────────────────────────────────────────────────────────────┘
```

### User Facing Changes

See #15607 for screenshots of translation, transcription, foia, and ihp admin actions

#### Schedule hearing admin action

Step|Shot
---|---
New Action|Screen Shot 2020-12-15 at 9 27 35 AM
New Modal|Screen Shot 2020-12-15 at 9 27 56 AM
Validation|Screen Shot 2020-12-15 at 9 27 49 AM
Success banner and New task|Screen Shot 2020-12-15 at 9 28 13 AM
Cannot create dupe|Screen Shot 2020-12-15 at 9 50 26 AM

#### Complete task

step|shot
---|---
Modal|Screen Shot 2020-12-15 at 9 30 26 AM
Banner|Screen Shot 2020-12-15 at 9 30 50 AM

### CODE REVIEW

File: app/models/concerns/cavc_admin_action_concern.rb [hunk 1]
```diff
@@ -19,4 +19,19 @@ def creating_from_cavc_workflow?(user, parent_task)
...
+
+    if assigned_to.is_a?(Organization) && appeal.tasks.open.where(type: type, assigned_to: assigned_to).any?
+      fail(
+        Caseflow::Error::DuplicateOrgTask,
+        docket_number: appeal.docket_number,
+        task_type: self.class.name,
+        assignee_type: assigned_to.class.name
+      )
+    end
+  end
```
Comment (hschallhorn) [inline app/models/concerns/cavc_admin_action_concern.rb:36]: Added some org task validation, specifically for schedule hearing tasks. Or standard validation https://github.com/department-of-veterans-affairs/caseflow/blob/04d89ceff826300ff7427db3e5a9cb6c314901cb/app%2Fmodels%2Ftask.rb#L278-L295
does not catch dupe schedule hearing tasks as each will have a different hearing task parent. This is not added directly to schedule hearing tasks as there are certain workflows (like postponing a hearing) where a schedule hearing task will be open at the same time as another (briefly, like how when we reassign a task we create a new task before we cancel the old task). Adding to the concern allows us to reuse the `creating_from_cavc_workflow?` check before performing the validation.

File: app/models/tasks/cavc_remand_processed_letter_response_window_task.rb [hunk 2]
```diff
@@ -11,7 +11,12 @@
 # Expected assigned_to.type: CavcLitigationSupport
 
 class CavcRemandProcessedLetterResponseWindowTask < Task
-  validates :parent, presence: true, parentTask: { task_type: CavcTask }, on: :create
+  VALID_PARENT_TYPES = [
+    CavcTask,
+    CavcRemandProcessedLetterResponseWindowTask
+  ].freeze
+
+  validates :parent, presence: true, parentTask: { task_types: VALID_PARENT_TYPES }, on: :create
```
Comment (hschallhorn) [inline app/models/tasks/cavc_remand_processed_letter_response_window_task.rb:19]: adds `CavcRemandProcessedLetterResponseWindowTask` as a valid task type so we can have an org and child task of the same type

File: app/models/tasks/cavc_remand_processed_letter_response_window_task.rb [hunk 3]
```diff
@@ -35,12 +40,35 @@ def default_instructions
...
 
-  USER_ACTIONS = [
-    Constants.TASK_ACTIONS.TOGGLE_TIMED_HOLD.to_h
+  TASK_ACTIONS = [
+    Constants.TASK_ACTIONS.MARK_COMPLETE.to_h,
+    Constants.TASK_ACTIONS.SEND_TO_TRANSLATION_BLOCKING_DISTRIBUTION.to_h,
+    Constants.TASK_ACTIONS.SEND_TO_TRANSCRIPTION_BLOCKING_DISTRIBUTION.to_h,
+    Constants.TASK_ACTIONS.SEND_TO_PRIVACY_TEAM_BLOCKING_DISTRIBUTION.to_h,
+    Constants.TASK_ACTIONS.SEND_IHP_TO_COLOCATED_BLOCKING_DISTRIBUTION.to_h,
+    Constants.TASK_ACTIONS.SEND_TO_HEARINGS_BLOCKING_DISTRIBUTION.to_h
```
Comment (hschallhorn) [inline app/models/tasks/cavc_remand_processed_letter_response_window_task.rb:49]: woooo admin actions!

File: app/models/tasks/schedule_hearing_task.rb [hunk 4]
```diff
@@ -23,6 +23,8 @@
 # If completed, an AssignHearingDispositionTask is created as a child of HearingTask.
 ##
 class ScheduleHearingTask < Task
+  include CavcAdminActionConcern
```
Comment (hschallhorn) [inline app/models/tasks/schedule_hearing_task.rb:26]: This allows us to directly create schedule hearing tasks as a child of a distribution task with the overridden verify user can create https://github.com/department-of-veterans-affairs/caseflow/pull/15694/files#diff-39f1b097dff2eecc7442f185cae27ff212beb8814090c63569060164996a8a9dL9. Also gives us a custom verify org task uniq to make sure we cannot create dupes https://github.com/department-of-veterans-affairs/caseflow/pull/15694/files#diff-39f1b097dff2eecc7442f185cae27ff212beb8814090c63569060164996a8a9dR25

File: app/repositories/task_action_repository.rb [hunk 5]
```diff
@@ -295,7 +299,8 @@ def change_task_type_data(task, user = nil)
     COMPLETE_TASK_MODAL_BODY_HASH = {
       NoShowHearingTask: COPY::NO_SHOW_HEARING_TASK_COMPLETE_MODAL_BODY,
       HearingAdminActionTask: COPY::HEARING_SCHEDULE_COMPLETE_ADMIN_MODAL,
-      SendCavcRemandProcessedLetterTask: COPY::SEND_CAVC_REMAND_PROCESSED_LETTER_TASK_COMPLETE_MODAL_BODY
+      SendCavcRemandProcessedLetterTask: COPY::SEND_CAVC_REMAND_PROCESSED_LETTER_TASK_COMPLETE_MODAL_BODY,
+      CavcRemandProcessedLetterResponseWindowTask: COPY::CAVC_REMAND_LETTER_RESPONSE_TASK_COMPLETE_MODAL_BODY
```
Comment (hschallhorn) [inline app/repositories/task_action_repository.rb:303]: custom copy for the complete task modal

File: client/app/queue/QueueApp.jsx [hunk 6]
```diff
@@ -685,6 +685,12 @@ class QueueApp extends React.PureComponent {
                 }`}
                 render={this.routedAssignToSingleTeam}
               />
+              <Route
+                path={`/queue/appeals/:appealId/tasks/:taskId/${
+                  TASK_ACTIONS.SEND_TO_HEARINGS_BLOCKING_DISTRIBUTION.value
+                }`}
+                render={this.routedAssignToSingleTeam}
+              />
```
Comment (hschallhorn) [inline client/app/queue/QueueApp.jsx:693]: New route for the "Assign schedule hearing to hearings team" action and modal

File: client/constants/TASK_ACTIONS.json [hunk 7]
```diff
@@ -310,6 +310,11 @@
     "value": "modal/assign_ihp_to_colocated_team",
     "func": "assign_ihp_to_colocated_blocking_distribution_data"
   },
+  "SEND_TO_HEARINGS_BLOCKING_DISTRIBUTION": {
+    "label": "Assign Schedule Hearing task to Hearings team",
+    "value": "modal/assign_schedule_hearing_to_hearings_team",
+    "func": "assign_schedule_hearing_to_hearings_blocking_distribution_data"
+  },
```
Comment (hschallhorn) [inline client/constants/TASK_ACTIONS.json:317]: New task action for the "Assign schedule hearing to hearings team"

File: app/repositories/task_action_repository.rb [hunk 8]
```diff
@@ -229,6 +229,10 @@ def assign_ihp_to_colocated_blocking_distribution_data(task, _user = nil)
       cavc_add_blocking_distrbution_admin_action_data(task, Colocated.singleton, IhpColocatedTask)
     end
 
+    def assign_schedule_hearing_to_hearings_blocking_distribution_data(task, _user = nil)
+      cavc_add_blocking_distrbution_admin_action_data(task, Bva.singleton, ScheduleHearingTask)
+    end
```
Comment (hschallhorn) [inline app/repositories/task_action_repository.rb:234]: New data for the "Assign schedule hearing to hearings team" action and modal

File: app/controllers/tasks_controller.rb [hunk 9]
```diff
@@ -21,6 +21,7 @@ class TasksController < ApplicationController
     ChangeHearingDispositionTask: ChangeHearingDispositionTask,
     ColocatedTask: ColocatedTask,
     CavcPoaClarificationTask: CavcPoaClarificationTask,
+    CavcRemandProcessedLetterResponseWindowTask: CavcRemandProcessedLetterResponseWindowTask,
```
Comment (hschallhorn) [inline app/controllers/tasks_controller.rb:24]: Allows us to assign and reassign `CavcRemandProcessedLetterResponseWindowTask`s to users

File: spec/feature/queue/cavc_task_queue_spec.rb [hunk 10]
```diff
@@ -161,56 +161,62 @@
     end
   end
```
Comment (hschallhorn) [inline spec/feature/queue/cavc_task_queue_spec.rb:163]: : ( Real ugly diff. Breaks out all the assigning, reassigning, and shared "send to team" workflows into shared examples.

File: spec/models/concerns/cavc_admin_action_concern_spec.rb [hunk 11]
```diff
@@ -32,4 +34,28 @@ class TestTask < Task
...
+
+      before { ScheduleHearingTask.create!(assigned_by: assigner, appeal: parent_task.appeal, parent: parent_task) }
+
+      context "when creating from a cavc workflow" do
+        it "fails validation" do
+          expect { subject }.to raise_error(Caseflow::Error::DuplicateOrgTask)
+        end
+      end
+
+      context "when creating from a non cavc workflow" do
```
Comment (yoomlam) [inline spec/models/concerns/cavc_admin_action_concern_spec.rb:52]:
```suggestion
      context "when creating from a non-cavc workflow" do
```

File: spec/models/concerns/cavc_admin_action_concern_spec.rb [hunk 12]
```diff
@@ -32,4 +34,28 @@ class TestTask < Task
...
+        let(:assigner) { create(:user) }
+
+        it "does not fail validation" do
+          expect { subject }.to_not raise_error
```
Comment (yoomlam) [inline spec/models/concerns/cavc_admin_action_concern_spec.rb:56]: Add a check for the newly created `ScheduleHearingTask`?

File: spec/models/concerns/cavc_admin_action_concern_spec.rb [hunk 13]
```diff
@@ -32,4 +34,28 @@ class TestTask < Task
...
+    context "ScheduleHearingTask" do
+      let(:assigner) { cavc_user }
+
+      subject { ScheduleHearingTask.create!(assigned_by: assigner, appeal: parent_task.appeal, parent: parent_task) }
+
+      before { ScheduleHearingTask.create!(assigned_by: assigner, appeal: parent_task.appeal, parent: parent_task) }
+
+      context "when creating from a cavc workflow" do
+        it "fails validation" do
+          expect { subject }.to raise_error(Caseflow::Error::DuplicateOrgTask)
```
Comment (yoomlam) [inline spec/models/concerns/cavc_admin_action_concern_spec.rb:48]: Add a check to make sure only one open `ScheduleHearingTask` exists?

**Review by yoomlam — COMMENTED** [review]
Whew! Tested and reviewed. Left some comments.

File: app/models/concerns/cavc_admin_action_concern.rb [hunk 14]
```diff
@@ -13,10 +18,25 @@ def verify_user_can_create!(user, parent_task)
     def creating_from_cavc_workflow?(user, parent_task)
       return true if parent_task&.type == DistributionTask.name && (
                      CavcLitigationSupport.singleton.user_is_admin?(user) ||
-                     SendCavcRemandProcessedLetterTask.open.where(assigned_to: user, appeal: parent_task.appeal).exists?
+                     parent_task.appeal.tasks.open.where(type: CAVC_USER_TASK_TYPES, assigned_to: user).exists?
                    )
 
       false
```
[the remark this answers is not in the document]
Comment (hschallhorn) [reply app/models/concerns/cavc_admin_action_concern.rb:24]: true true true. Updating!

**Review by yoomlam — APPROVED** [review]
🥇 🚀

**Pull request merged**